doc/notes/188_enhance_logging_ant.adoc

```diff
@@ -0,0 +1,245 @@
+  performance of the Protocol Verifier application itself (benchmarking).
+* Munin software shall be instrumented sparingly with debug statements
+  adequate for defect isolation (debugging).
+* The volume of production-level logging shall not exceed the volume of
+  production audit event traffic.
+
+=== 4.2 Breakdown of Application Traffic
+
+The Protocol Verifier monitors devices through telemetry.  There are two
+levels of detail in this telemetry, that of a _job_ and that of an
+_event_.  In general, the user is interested in the status of a job.
+Events are interesting only when they are erroneous.  Summaries of
+aggratated events are interesting in terms of throughput.
+
+=== 4.3 Job Logging
+
+The primary deliverable log entry is the final status of a job.  A job
+completes with status _successful_, _failed_ or _alarm_.  Together with
+the job type (name) and identifier, this is the key status provided by the
+Protocol Verifier.  Failed and alarmed jobs must also report internal
+state satisfactory to understanding failure modes in application-level
+jobs in the system being monitored.  Canonical logging practices shall be
+applied to facilitate problem determination for failed and alarmed jobs.
+
+To provide canonical log messages once and only once per job, information
+about the job and events will be accumulated in instance attributes of the
+modelled artefacts.  At the end of the job, the accumulated information
+will be structured and delivered to the logger.
+
+==== 4.3.1 Job-Level Statistics for Successful Jobs
+
+For successful jobs, the following shall be reported:
+
+* job type (name)
+* job identifier
+* event count
+
+This job status report shall be logged exactly once at completion of the
+job.  This information shall be contained in a single structured log
+message.
+
+The recommended location of this logging is in Sequence Verification.
+
+==== 4.3.2 Job-Level Statistics for Failures and Alarms
+
+For failed and alarmed jobs, the following statistics are required:
+
+* job type (name)
+* job identifier
+* event count
+* failure/alarm mode
+* event trace
+  ** event type (name)
+  ** event identifier
+  ** ordering
+
+=== 4.4 Event Logging
+
+Logging at the event level is largely infeasible and unnecessary in
+production configurations.  However, in anticipation of performance
+benchmarking, some event-level statistics shall be gathered in summary
+format.
+
+==== 4.4.1 Event Statistics
+
+=== 4.5 Protocol Verifier Statistics
+
+Several parameters are useful in the maintenance and support of the
+Protocol Verifier.
+
+The following statistics are reported as they happen.
+
+* worker registered
+* worker deregistered
+* unassigned job count exceeded
+  ** job count exceeded is reported once upon occurrence and then only
+     reported again if the job count goes below the threshold.
+
+The following statistic are reported periodically.
+
+* event count since start
+* event throughput
+* worker count
```

Comment:
Thinking about the dashboard, would the **average** event throughput be determined by summing a set of the periodic event throughput values and dividing by the number of periods?